### DISCUSSION TITLE
Can we change background color?

### DISCUSSION THREAD

**Discussion by keiohta** [haosulab/SAPIEN#81] [Q&A] [answered]
Hi author, would it be possible to change the background color to white (transparent)? Thanks!

**Comment by fbxiang** [comment, marked as answer]
You can customize the shader following these steps:
1. make a copy of the vulkan_shader folder
2. pick the shader pack you want to use (I believe the default is the ibl)
3. Modify the background color used in `deferred.frag` (this line: https://github.com/haosulab/SAPIEN/blob/6bc3f4e2be910199b793f185aea5791d9f193e4c/vulkan_shader/ibl/deferred.frag#L242) For example, change it to `outLighting=vec4(1,1,1,0)`
4. call `sapien.VulkanRenderer.set_viewer_shader_dir` and/or `sapien.VulkanRenderer.set_camera_shader_dir` before creating the viewer or the camera. Give it the directory of the modified ibl folder.

**Reply by keiohta** [reply]
Thanks! I successfully changed the background.

Actually, I failed to change the shader directory, so I directly modified [the default deferred.frag](https://github.com/haosulab/SAPIEN/blob/master/vulkan_shader/default_viewer/deferred.frag#L199) and the background becomes white.

Just for your information, I only added

```
sapien.VulkanRenderer.set_viewer_shader_dir("ibl")
sapien.VulkanRenderer.set_camera_shader_dir("ibl")
```

before creating `engine` and `renderer`, and encountered an error

```
  File "/home/ohta/anaconda3/envs/sapien/lib/python3.8/site-packages/sapien/utils/viewer.py", line 1855, in render
    self.window.render(
RuntimeError: custom sampler "Environment" is not set in the renderer
```

The SAPIEN version I use is `1.1.1`.

**Reply by fbxiang** [reply]
You are right. For future reference, SAPIEN 2 has removed the default shader and defaults to ibl. The "Environment" problem is also fixed at some point(dev branch).
